### DISCUSSION TITLE
What happen when two formulae use the same command

### DISCUSSION THREAD

**Discussion by quackerex** [Homebrew/discussions#4908] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

What happen when the command is same for different formulas? 

Brew documentation mention this but it seem about name conflict rather the command conflict

> When Homebrew already has a formula called foo we typically do not accept requests to replace that formula with something else also named foo. This is to avoid both confusing and surprising users’ expectations.

For example if there is two formulae named, `foo` and `better-foo` but use the same command `foo` how does brew handle it?

When I say 'command' I am referring to [1] part
```
┌1┐
git checkout master
```

**Comment by alebcay** [comment, marked as answer]
As a general rule of thumb, we may try to install a binary/command by a different name for one of the formulae. One example of this is [in the `fcrackzip` formula, where `zipinfo` is renamed to `fcrackzipinfo`](https://github.com/Homebrew/homebrew-core/blob/b38f5c762797299079a27419fecd765164f2e917/Formula/f/fcrackzip.rb#L35) to avoid conflicts with `zipinfo` provided from `unzip`.

In cases where it is not sensible to rename the binary/command in either of the formulae, the `conflicts_with` stanza can be used to indicate that the two formulae can't be installed simultaneously (`brew install` would otherwise fail on the link step, refusing to overwrite the symlink to binary of the other formula). One example of this is the conflict marked between [`yq`](https://github.com/Homebrew/homebrew-core/blob/b38f5c762797299079a27419fecd765164f2e917/Formula/y/yq.rb#L27) and [`python-yq`](https://github.com/Homebrew/homebrew-core/blob/b38f5c762797299079a27419fecd765164f2e917/Formula/y/yq.rb#L27). Note that the conflict must be denoted on both sides (or, in the case of a three-way conflict, all three formulae must declare a conflict with the other two).
